### PR TITLE
[SwiftLanguageRuntime] Handle Optional correctly when resolving promi…

### PR DESCRIPTION
**Pull request by dcci**
…ses.

<rdar://problem/45049135>

### CODE REVIEW

**Comment by dcci** [comment]
@swift-ci test and merge

File: packages/Python/lldbsuite/test/lang/swift/protocol_optional/TestSwiftProtocolOptional.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+#
+# See https://swift.org/LICENSE.txt for license information
+# See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+#
+# ------------------------------------------------------------------------------
+import lldbsuite.test.lldbinline as lldbinline
+from lldbsuite.test.decorators import *
+
+lldbinline.MakeInlineTest(__file__, globals(),
+                          decorators=[swiftTest,skipUnlessDarwin])
```
Comment (adrian-prantl) [inline packages/Python/lldbsuite/test/lang/swift/protocol_optional/TestSwiftProtocolOptional.py]: What is Darwin-specific about this test?

Comment (dcci) [reply packages/Python/lldbsuite/test/lang/swift/protocol_optional/TestSwiftProtocolOptional.py]: like, nothing. Cut&paste failure.

Comment (dcci) [reply packages/Python/lldbsuite/test/lang/swift/protocol_optional/TestSwiftProtocolOptional.py]: Should be fixed now.

**Comment by dcci** [comment]
@swift-ci test and merge

**Pull request merged**